File: script/model/tienda.js
class Tienda{
    constructor(nombre, correo, direccion, password){
        this.id = 0;
        this.nombre = nombre;
        this.correo = correo;
        this.direccion = direccion;
        this.password = password;
    }

    loadFromParsedJson(object){
        this.id = object.id;
        this.nombre = object.nombre;
        this.correo = object.correo;
        this.direccion = object.direccion;
        this.password = object.password;
    }

    loadFromJson(json){
        var parsedTienda = JSON.parse(json);
        this.id = parsedTienda.id;
        this.nombre = parsedTienda.nombre;
        this.correo = parsedTienda.correo;
        this.direccion = parsedTienda.direccion;
        this.password = parsedTienda.password;
    }

    stringify(){
        return JSON.stringify(this);
    }
}


export { Tienda };

File: bnpt/src/main/java/com/bnpt/model/entities/Tienda.java
package com.bnpt.model.entities;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "tiendas")
public class Tienda {
	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

	@Column(name = "nombre", nullable = false, length = 50)
    private String nombre;
    
    @Column(name = "direccion", nullable = false, length = 100)
    private String direccion;
    
    @Column(name = "correo", nullable = false, length = 50)
    private String correo;

    @Column(name = "password", nullable = false, length = 256)
	private String password;

	public int getId() {
		return this.id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getNombre() {
		return this.nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public String getDireccion() {
		return this.direccion;
	}

	public void setDireccion(String direccion) {
		this.direccion = direccion;
	}

	public String getCorreo() {
		return this.correo;
	}

	public void setCorreo(String correo) {
		this.correo = correo;
	}

	public String getPassword() {
		return this.password;
	}

	public void setPassword(String password) {
		this.password = password;
	}
}